### PR TITLE
Fix init snowflake table issue for init kafka offset not zero

### PR DESCRIPTION
**Pull request by KingNF**
I found a issue about the connector.

**Situation :** when kafka topic has expired data , it's partition offset not started with zero. 

**Kafka connector config:**  snowpipe streaming + schema detection

**Problem:**  the kafka connector will only create snowflake table  not  ingest data to snowflake table when first run the connector,  must restart the  connector can ingest data normally.

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java [hunk 1]
```diff
@@ -68,6 +68,7 @@ public class TopicPartitionChannel {
 
   private static final long NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE = -1L;
 
+  private long initKafkaOffset = 0L ;
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:71]: can you rename this to `firstSeenKafkaOffset`

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java [hunk 2]
```diff
@@ -319,6 +321,14 @@ public void insertRecordToBuffer(SinkRecord kafkaSinkRecord) {
    */
   private boolean shouldIgnoreAddingRecordToBuffer(SinkRecord kafkaSinkRecord) {
     if (this.isOffsetResetInKafka.get()) {
+      if (offsetPersistedInSnowflake.get() == -1L && kafkaSinkRecord.kafkaOffset() == initKafkaOffset) {
+        LOGGER.info(
+                "First insert to table, get the desired init offset:{} from Kafka, we can add this offset to buffer for channel:{}",
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java:326]: `First insert to table, first seen offset from Kafka:{}, we can add this offset to buffer for channel:{}",`

**Review by sfc-gh-japatel — COMMENTED** [review]
Thank a lot for the contribution @KingNF , can you also add a test? 
FYI, first time contributors are required to sign a CLA. Please send an email address where we can forward the CLA.

**Comment by KingNF** [comment]
Thank you for your review @sfc-gh-japatel . I'm sorry for updating the code so late. I have updated it according to your comment and added a new test. If there are still areas that need to be modified, please let me know, and this is my email address: fcyangwork@gmail.com

**Comment by sfc-gh-japatel** [comment]
I guess this can be closed since @sfc-gh-tzhang merged a change #587 

Please confirm Toby. 

For future reference, we have also added CLA bot now which can be used to sign the CLA here on github. 

Once you do that. we will run tests and push (if approved)

**Comment by sfc-gh-tzhang** [comment]
This should be fixed with 1.9.2, please give it a try!

**Pull request closed without merging**